zebra-state/src/service/finalized_state.rs

```diff
@@ -206,6 +228,10 @@ impl FinalizedState {
         let committed_tip_hash = self.db.finalized_tip_hash();
         let committed_tip_height = self.db.finalized_tip_height();
 
+        // Save blocks to elasticsearch if the feature is enabled.
+        #[cfg(feature = "elasticsearch")]
+        self.elasticsearch(&finalized.block, &committed_tip_height);
```

Comment:
Could this call be moved to the `if result.is_ok()` block.

Reply:
Moved at https://github.com/ZcashFoundation/zebra/pull/6274/commits/bf7b0fb07c957adc626a434f49bc34d79e221b8a